### ISSUE TITLE
Suggestions from ClassProvider are being dropped because they share the same text.

### ISSUE THREAD

**Issue by ghost** [php-integrator/atom-autocompletion#15]
Apparently, autocomplete-plus filters out valid suggestions if they share the same `text` property.
##

**Comment by Gert-dev** [comment]
See also https://github.com/atom/autocomplete-plus/issues/615.

**Comment by Gert-dev** [comment]
Note: This currently applies to the class provider, but it will also apply to the `MemberProvider` starting from version 0.9.0, for example:

```php
/**
  * @return \Iterator|\IteratorIterator
  */
public function foo()
{
    $this->foo()-> // Problem.
}
```

In this case, there will be a couple of duplicate members (e.g. from the Iterator interface via both type branches). Because autocomplete-plus naturally filters out items with the same text, the duplicates won't actually show. However, this is a problem, as it is very possible that there are two branches that both share a method `bar` where they have different parameters and originate from different classlikes. In that case, we _would_ want both to be listed.

**Comment by hultberg** [comment]
Any updates on this @Gert-dev, this is a quite annoying issue in a large php project.

**Comment by Gert-dev** [comment]
@hultberg I agree wholeheartedly and am also bothered by this every day. I can't count the times I have to write a generic class name such as `Request`, have about four matches and have to write the import manually because autocomplete-plus decided to select the one I don't want.

Unfortunately, there is very little I can do to fix this and every solution on my end is some sort of compromise:
* I can always complete fully qualified class names, but that leads to verbose code and makes imports mostly useless.
* I can insert fully qualified class names, add an import, then modify the completed code to remove part of it. This will fix the issue, but it won't work properly for multiple cursors.

As I noted in my bug report to autocomplete-plus: there should be a way to have the same suggestion multiple times if the metadata is different or at least have some way to indicate this by specifying some sort of `noMerge` flag in the completion.

Sadly, from what I've gathered, and judging from the rate pull requests from external contributors get accepted in Atom core and packages, the Atom team is very busy (possibly even overwhelmed) by the amount of issues and pull requests across their various projects, so I'm not even sure it would help to make one.

**Comment by hultberg** [comment]
@Gert-dev Heads up: atom/autocomplete-plus#893

**Comment by hultberg** [comment]
Just tested it out, works!
![2017-09-26_855x249](https://user-images.githubusercontent.com/1257913/30849499-d5da5f76-a2a2-11e7-9e04-91c28c4b0b3a.png)

**Comment by Gert-dev** [comment]
@hultberg _\*Lets out a sigh in relief.\*_

This must have been one of _the_ single most annoying issues so far. I underestimated how often you have classes with the same name in a large code base.

Did it require any changes or did it just work out of the box? In any case, I'll leave this issue open until the fix has landed in a stable Atom release.

**Comment by hultberg** [comment]
@hultberg I did `apm install autocomplete-plus` and downloaded the latest autocomplete version, and it just worked. This version that fixes this issue in autocomplete-plus is a part of atom beta, so it will land in stable soon.

**Comment by Gert-dev** [comment]
Closing this as it seems to _finally_ be fixed in Atom 1.21 🎉 .
